ddtrace/utils/formats.py

```diff
@@ -97,7 +99,25 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtags = tags_str.count(":")
```

Comment:
WDYT of
~~~ python
    n = tags.count(":")
    for s in ", ":
        if n == sum(":" in _ for _ in tags.split(s)):
            sep = s
            break
    else:
        log.error(...)
        return parsed_tags
~~~

Reply:
In fact, I'm not sure this is even correct. Perhaps the better thing to do is
~~~ python
    n = tags.count(":")
    for s in ", ":
        if all(_.count(":") == 1 for _ in tags.split(s)):
            sep = s
            break
    else:
        log.error(...)
        return parsed_tags
~~~

Reply:
The only (minor) issue with the `all(...)` solution is that it would consider `key1,key2,key3` an error having to do with determining the separator when it's that the tags themselves are not well-formed.

Reply:
Ok, forget what I said, the original approach is indeed fine! Sorry for all the confusion! 😞